grakn-core/src/main/java/ai/grakn/concept/ConceptId.java

```diff
@@ -21,34 +21,45 @@
 import ai.grakn.GraknTx;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonValue;
-import com.google.auto.value.AutoValue;
 
 import javax.annotation.CheckReturnValue;
 import java.io.Serializable;
 
 /**
- * <p>
- *     A Concept Id
- * </p>
+ * A class which represents an id of any {@link Concept} in the {@link GraknTx}.
+ * Also contains a static method for producing concept IDs from Strings.
  *
- * <p>
- *     A class which represents an id of any {@link Concept} in the {@link GraknTx}.
- *     Also contains a static method for producing concept IDs from Strings.
- * </p>
- *
- * @author Filipe Peliz Pinto Teixeira
+ * @author Grakn Warriors
  */
-@AutoValue
-public abstract class ConceptId implements Comparable<ConceptId>, Serializable {
-    private static final long serialVersionUID = -1723590529071614152L;
+public class ConceptId implements Comparable<ConceptId>, Serializable {
+
+    private final String value;
```

Comment:
I think if you make it @Nullable autovalue will make it a null by default which is the behaviour you want. Then you could add a create() method to get the non-argument constructor behaviour.

Reply:
Still doesn't work for the OLAP framework, i've tried